backend/core/src/test/kotlin/com/ritense/valtimo/security/x509/SubjectDnX509PrincipalExtractorVulnerabilityTest.kt

```
@@ -0,0 +1,145 @@
+
+    @ParameterizedTest(name = "[{index}] DN: {0}")
+    @MethodSource("malformedDNsWithExpectedResults")
+    @DisplayName("Malformed DN should not allow impersonation")
+    fun `malformed DN should not allow impersonation`(
+        subjectDN: String,
+        forbiddenPrincipal: String,
+        description: String
+    ) {
+        val cert = try {
+            X509TestCertificateGenerator.createCertificate(subjectDN)
+        } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)